modules/post/windows/gather/forensics/memorydump.rb

```diff
@@ -0,0 +1,230 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+#
+# This Module has an embred Belkasoft RAM Capturer version of 2018-08
+# Available at https://belkasoft.com/ram-capturer
+##
+
+#
+# Forensic memory dump using Belkasoft RAM Capturer
+#
+# Helvio Junior (M4v3r1cK) m4v3r1ck.hjr [at] gmail.com
+#    https://www.helviojunior.com.br
+#
+# Other modules repository: https://github.com/helviojunior/metasploit_modules
+#
+# Standard Library
+#
+require 'tmpdir'
+
+
+class MetasploitModule < Msf::Post
+  include Msf::Post::File
+  include Msf::Post::Windows::UserProfiles
+
+  def initialize(info={})
+    super( update_info( info,
+      'Name'          => 'Windows Gather Memory Dump Imaging',
+      'Description'   => %q{This module will perform a memory dump of remote machine},
+      'License'       => MSF_LICENSE,
+      'Platform'      => ['win'],
+      'SessionTypes'  => ['meterpreter'],
+      'Author'        => ['Helvio Junior (M4v3r1cK) <m4v3r1ck.hjr[at]gmail.com>']
+    ))
+    register_options(
+      [
+        OptString.new('RAMCAPTURE_PATH',[true,'Path of Belkasoft RAM Capturer',''])
+      ])
+  end
+
+  def run
+    vprint_status("Determining session platform, type and privileges")
+    case session.platform
+    when 'windows'
+      if session.type != "meterpreter"
+        print_error "Only meterpreter sessions are supported on Windows hosts"
+        return
+      end
+      @platform = :windows
+    else
+      print_error("Unsupported platform: #{session.platform}")
+      return
+    end
+
+    unless got_root
```

Comment:
The `lib/msf/core/post/windows/priv.rb` lib provides `is_admin?` and `is_system?` methods you can use here, rather than implementing your own `got_root` method.